extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -59,19 +96,32 @@ export async function runRemoteQuery(cliServer: cli.CodeQLCliServer, credentials
   const query = await fs.readFile(queryFile, 'utf8');
 
   const repositoriesFile = queryFile.substring(0, queryFile.length - '.ql'.length) + '.repositories';
-  if (!(await fs.pathExists(repositoriesFile))) {
-    void showAndLogErrorMessage(`Missing file: '${repositoriesFile}' to specify the repositories to run against. This file must be a sibling of ${queryFile}.`);
-    return;
-  }
+  let ref: string | undefined;
+  let language: string | undefined;
+  let repositories: string[] | undefined;
+
+  // If the user has an explicit `.repositories` file, use that.
+  // Otherwise, prompt user to select repositories from the `codeQL.remoteRepositoryLists` setting.
+  if (await fs.pathExists(repositoriesFile)) {
+    void logger.log(`Found '${repositoriesFile}'. Using information from that file to run ${queryFile}.`);
 
-  const config = yaml.safeLoad(await fs.readFile(repositoriesFile, 'utf8')) as Config;
+    const config = yaml.safeLoad(await fs.readFile(repositoriesFile, 'utf8')) as Config;
 
-  const ref = config.ref || 'main';
-  const language = config.language || await findLanguage(cliServer, uri);
-  const repositories = config.repositories;
+    ref = config.ref || 'main';
+    language = config.language || await findLanguage(cliServer, uri);
+    repositories = config.repositories;
+  } else {
+    ref = 'main';
+    language = await findLanguage(cliServer, uri);
+    repositories = await getRepositories();
```

Comment:
Since `findLanguage` can take a while to run and `getRepositories` is blocked by a user event, here's a trick to get them to run in "parallel":

```suggestion
    [ language, repositories ] = await Promise.all([findLanguage(cliServer, uri), getRepositories()]);;
```

`Promise.all` accepts an array of promises and resolves when all of its arguments resolve. Or, it rejects if any one of them reject. The return value is an array containing all of the resolved values of each promise.